### ISSUE TITLE
Patterns not working properly when being extended inside a component

### ISSUE THREAD

**Issue by omarkhatibco** [chakra-ui/panda#1082]
### Description

Maybe the title is not the best but let's have some example here.

so I used the `Circle` pattern to make a skeleton component that accept the same props expect it has extra bgColor

now the code of the component look like this 

```ts
import { forwardRef } from 'react';
import { Circle, CircleProps } from '../styled-system/jsx';

export const SkeletonCircle = forwardRef<HTMLDivElement, CircleProps>(
  (props, ref) => {
    return <Circle ref={ref} bgColor='surface.container.highest' {...props} />;
  },
);
```

now when I use the component inside my code like this 

```ts
  <SkeletonCircle size='100px' />
```

now the styles for the this component is unfortunately is not generated specifically `width` & `height`

### Link to Reproduction

https://codesandbox.io/p/github/omarkhatibco/pandacss-test/main?file=%2Fsrc%2FApp.tsx%3A1%2C52&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clke5o5ji000g3b6na2318y48%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clke5o5ji000b3b6nd4kxst45%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clke5o5ji000f3b6n7t7f2rnn%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clke5o5ji000d3b6nirf5f9qh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clke5o5ji000b3b6nd4kxst45%2522%253A%257B%2522id%2522%253A%2522clke5o5ji000b3b6nd4kxst45%2522%252C%2522activeTabId%2522%253A%2522clke5vqai00l33b6n05xitvu1%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clke5o5ji000a3b6nsp0rse5z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftsconfig.node.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522id%2522%253A%2522clke5vqai00l33b6n05xitvu1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clke5o5ji000d3b6nirf5f9qh%2522%253A%257B%2522id%2522%253A%2522clke5o5ji000d3b6nirf5f9qh%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clke5orze01953b6nxmr0qlgi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clke5orze01953b6nxmr0qlgi%2522%257D%252C%2522clke5o5ji000f3b6n7t7f2rnn%2522%253A%257B%2522id%2522%253A%2522clke5o5ji000f3b6n7t7f2rnn%2522%252C%2522activeTabId%2522%253A%2522clke5oo4100xt3b6no30sbz1t%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clke5o5ji000e3b6nanqwvsnb%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clke5o5rl0003h3gn00u59cq5%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clke5oo4100xt3b6no30sbz1t%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Panda CSS Version

0.7

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

**Comment by astahmer** [comment]
so basically the issue you mention is not being able to wrap patterns in custom components (and you're right, you can't atm)

some solutions that I can think of :
1. -  having a similar `jsx` property, [as recipes currently have](https://panda-css.com/docs/concepts/recipes#advanced-jsx-tracking), that you (the user) would have to fill by yourself to make the pattern components known at build-time (= without making the static extractor performance suffer by having to scan multiple times app code)

so, in your example, it could look like:
```
// panda.config.ts
// ..
patterns: {
    extend: {
        Circle: {
             jsx: ["Circle", "SkeletonCircle"]
        }
   }
}

```

2. - having a similar `xxx.component()` HoC method (like the [`xxx.raw()` function that recently got added](https://github.com/chakra-ui/panda/pull/1069)), that would again just be an identity function and would only be used to mark things for the static extractor

so, in your example, it could look like:

```tsx
import { forwardRef } from 'react';
import { Circle, CircleProps } from '../styled-system/jsx';
import { circle } from '../styled-system/patterns';

export const SkeletonCircle = circle.component(forwardRef<HTMLDivElement, CircleProps>(
  (props, ref) => {
    return <Circle ref={ref} bgColor='surface.container.highest' {...props} />;
  },
));
```

---

I thought of looking for `{Pattern}Props` (like `CircleProps` in your example) in function definition generics and interpreting the `VariableDeclaration.Identifier` as a JSX pattern name (`SkeletonCircle` for you) 

But even tho this should be the most convenient but it would also probably be the most error prone as well 
It would work just fine for simple cases but that means we also have to care about `fowardRef` (but then people's own HoC functions will not work and that wouldn't be intuitive), or maybe they would use it in complex TS typings etc etc, so many possible cases that I'd rather not think about

---

> Another issue I just noticed, that mapping css props to different name is not working as well

so inside SkeletonText I map the skeletonHeight to css height prop on the Box and it's not working as well

this one is a known limitation. it should be documented in more places but this is mentioned [here for recipes](https://panda-css.com/docs/concepts/recipes#limitations), the same principle applies for all props (style props, pattern props, recipe props..)
basically for the same reasons mentioned above: keep the static extractor performance fast by not scanning app code multiple times

---

> Due to static nature of Panda, it's not possible to track the usage of the recipes in all cases. Here are some of use cases that Panda won't be able to track the usage of the recipe variants:

When you change the name of the variant prop in the JSX component

```tsx
const Button = ({ buttonSize, children }) => {
  return (
    <button {...props} className={button({ size: buttonSize })}>
      {children}
    </button>
  )
}
```

**Comment by omarkhatibco** [comment]
@astahmer ,

1. 
for solution number one with the `jsx` property, it could help but in my case, I use panda in a component library and the end application does not have access to the recipe/patterns, I don't know if the user would use the pattern in building their own components would extend in the same way.

I faced the same problem with recipes and converted all the components to a normal one to avoid such a conflict.


2.
Don't complicate yourself with such a function I know it's overhead work to get it to work with all the frameworks, but here I got a question, would `xx.raw()` function work if I passed it somehow to the component, or would it work if I passed the height to the `css` function?

3. 
another question in general about `staticCss` what about generating all the classes and then we let purgeCss remove the unused one(for sure I'm asking to so on the application level, not inside Panda)  would you consider this one a solution? or do you see any limitations here?

**Comment by anulman** [comment]
I'm also experiencing this issue. I like option 1 most; in my case, _all_ of the custom recipes I override are located in the same directory. If the library goes this route, would it be possible to specify a glob of files as a hint to the static extractor, vs. the names of every one of the exported components?
